src/main/java/be/syntra/devshop/DevshopFront/controllers/ProductController.java

```diff
@@ -51,14 +51,18 @@ public String displayProductOverview(Model model) {
     @GetMapping("/details/{id}")
     public String handleGet(@PathVariable Long id, Model model) {
         Product product =
-                (null == productListCacheService.findById(id))
+                isProductListCacheEmpty(id)
                         ? productService.findById(id)
                         : productListCacheService.findById(id);
         model.addAttribute("product", product);
         model.addAttribute("cart", cartService.getCart());
         return "product/productDetails";
     }
 
+    private boolean isProductListCacheEmpty(@PathVariable Long id) {
```

Comment:
ctrl-alt-M seems to be very thorough. Is @PathVariable really necessary in the boolean method?